src/test/java/com/adobe/cq/commerce/magento/graphql/QueryBuilderTest.java

```diff
@@ -191,4 +193,92 @@ public void testMutationRequest() throws Exception {
         Assert.assertEquals("John", customer.getFirstname());
         Assert.assertEquals("Doe", customer.getLastname());
     }
+
+    @Test
+    public void testCustomFieldTypes() throws Exception {
+        String expectedQuery = getResource("queries/category-with-custom-field.txt");
+        String jsonResponse = getResource("responses/category-with-custom-field.json");
+
+        // Category with custom fields of different primitve types
+        CategoryTreeQueryDefinition queryArgs = q -> q
+            .id()
+            .name()
+            .addCustomSimpleField("mystring")
+            .addCustomSimpleField("myinteger")
+            .addCustomSimpleField("mydouble")
+            .addCustomSimpleField("myboolean")
+            .addCustomSimpleField("myarray");
+
+        // Search parameters
+        CategoryArgumentsDefinition searchArgs = q -> q.id(3);
+
+        String queryString = Operations.query(query -> query.category(searchArgs, queryArgs)).toString();
+
+        // Check that the generated query matches the reference query
+        Assert.assertEquals(expectedQuery, queryString);
+
+        // Check that the reference response can be parsed and fields are properly set
```

Comment:
I would move this part in a separate test, since it's basically independent of the first part. But that's just personal preference :)